### DISCUSSION TITLE
Claims syntax for External Identity Management Plugin

### DISCUSSION THREAD

**Discussion by dridk** [minio/minio#20369] [Q&A]
I'm trying to create an external identity manager for authentification. But I can't specify the policy for the authentificate user. I don't know how to set the "claims" attribute in http response.

- In the minio configuration, I have defined:

```
MINIO_IDENTITY_PLUGIN_ROLE_POLICY=“diagnosis”` 
MINIO_IDENTITY_PLUGIN_URL="http://localhost:6000"
MINIO_IDENTITY_PLUGIN_ROLE_ID='custom-auth'
```

- I have a policy call `access-test`
- I created a python server to return the authentification validation. But I have no idea how to set the claims ?

```python
from flask import Flask , request
  app = Flask(__name__)
  @app.route(“/”, methods=[“POST”, “GET”, “PUT”])
  def test():

      return {
          “user": ‘sacha’,  # Is this a random string ? Is it important ? 
          “maxValiditySeconds":3600,
          “claims": { ‘policy’: 'access-test’}  # HOW TO SET CLAIMS ???
      }

```

From a terminal, I login with : 

```bash
 http --verify no -v POST :3015  Action==AssumeRoleWithCustomToken Token==$VAULT_TOKEN Version=='2011-06-15' DurationSeconds==1000 RoleArn=='arn:minio:iam:fr-west-1::role/idmp-custom-auth'
```

this command is valid and returns an access_key, a secret_key and a session_key. But when I use them, I don't have the policy  "access-test" attached. It looks like that only the policies defined in MINIO_IDENTITY_PLUGIN_POLICY are defined. My claims has no effect. 

Could you give me the right syntax for the claims ?

**Comment by dridk** [comment]
I am confused with these 2 documentations

https://min.io/docs/minio/linux/administration/identity-access-management/pluggable-authorization.html

https://min.io/docs/minio/linux/administration/identity-access-management/pluggable-authentication.html

**Reply by dridk** [reply]
Looks like you are using groups 
https://github.com/minio/minio/blob/d7ef6315aea1a72f0c55828b6d6a19f915382647/docs/iam/identity-manager-plugin.go#L49

**Comment by NotSoWell** [comment]
You must define a POST method which will take token from query parameters.
About `user` in return model - i think this field needed for audit or logging purposes.
About claims - see my original answer https://github.com/minio/minio/discussions/20106#discussioncomment-10599372
